### DISCUSSION TITLE
Use __future__.annotations

### DISCUSSION THREAD

**Discussion by karpetrosyan** [encode/httpx#3047] [Ideas]
We can consider using the most recent type annotations to avoid importing types such as `Tuple`, `List`, `Union`, and many more.

`Uvicorn` has already switched to the new typing style in [this pull request](https://github.com/encode/uvicorn/pull/2199/files), so I believe we can follow suit and use __future__.annotations in `httpx` and `httpcore`.

**Comment by Kludex** [comment]
Yes. 👍 

I'm doing this on Starlette now.

**Reply by karpetrosyan** [reply]
What do you think. Should we add a CI check for this change to prevent using the old typing style?
